proteuslib/unit_models/reverse_osmosis_1D.py

```diff
@@ -0,0 +1,800 @@
+
+    CONFIG.declare("area_definition", ConfigValue(
+            default=DistributedVars.uniform,
+            domain=In(DistributedVars),
+            description="Argument for defining form of area variable",
+            doc="""Argument defining whether area variable should be spatially
+    variant or not. **default** - DistributedVars.uniform.
+    **Valid values:** {
+    DistributedVars.uniform - area does not vary across spatial domain,
+    DistributedVars.variant - area can vary over the domain and is indexed
+    by time and space.}"""))
+
+    CONFIG.declare("property_package", ConfigValue(
+            default=None,
+            domain=is_physical_parameter_block,
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property calculations
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}"""))
+
+    CONFIG.declare("property_package_args", ConfigValue(
+            default={},
+            description="Arguments for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these.
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}"""))
+
+    CONFIG.declare(
+        "transformation_method",
+        ConfigValue(
+            default=useDefault,
+            description="Discretization method to use for DAE transformation",
+            doc="""Discretization method to use for DAE transformation. See Pyomo
+    documentation for supported transformations."""))
+
+    CONFIG.declare("transformation_scheme", ConfigValue(
+            default=useDefault,
+            description="Discretization scheme to use for DAE transformation",
+            doc="""Discretization scheme to use when transforming domain. See
+    Pyomo documentation for supported schemes."""))
+
+    CONFIG.declare("material_balance_type", ConfigValue(
+            default=MaterialBalanceType.useDefault,
+            domain=In(MaterialBalanceType),
+            description="Material balance construction flag",
+            doc="""Indicates what type of mass balance should be constructed,
+    **default** - MaterialBalanceType.useDefault.
+    **Valid values:** {
+    **MaterialBalanceType.useDefault - refer to property package for default
+    balance type
+    **MaterialBalanceType.none** - exclude material balances,
+    **MaterialBalanceType.componentPhase** - use phase component balances,
+    **MaterialBalanceType.componentTotal** - use total component balances,
+    **MaterialBalanceType.elementTotal** - use total element balances,
+    **MaterialBalanceType.total** - use total material balance.}"""))
+
+    CONFIG.declare("energy_balance_type", ConfigValue(
```

Comment:
Later on you set energy balance type to `None`. Either don't add it here as an option, or make use of this argument later (which would require supporting energy balances).

Reply:
For now, I will delete the option and add support for energy balances after I get other parts working.